### PR TITLE
NNotepad: UI update

### PR DESCRIPTION
**Pull request by ibelem**
Updated UI with following changes:

- Add navigation bar for logo, WebNN sample links, show generated code, show documentation and device type selector
- Add code editor to replace the <textarea> which can show code lines
- More style tuning

@inexorabletash @huningxin @Honry Please take a look and let me know if it's acceptable or not.

Screenshot 2024-06-12 094227

### CODE REVIEW

**Review by huningxin — COMMENTED** [review]
Does this change from "Nnodepad" to "NNotepad"? I cannot tell that from the diff.

**Comment by ibelem** [comment]
@huningxin There was the typo in the svg logo, changed the text from "NNodepad" to "NNotepad" now, please refer to the updated new screen shot as attached.

**Comment by huningxin** [comment]
The UI update looks good to me. I'll defer to @inexorabletash for decision. Thanks!

File: nnotepad/css/index.css
Comment (inexorabletash) [inline nnotepad/css/index.css:6]: These references to `../common` all need to be `../../common` to reach up out of `nnotepad/css`

File: nnotepad/css/index.css
Comment (inexorabletash) [inline nnotepad/css/index.css:188]: `"../common/css/fonts/victor-mono-v3-latin-regular.woff2"` 

should be

`"../../common/css/fonts/victor-mono-v3-latin/victor-mono-v3-latin-regular.woff2"`

(note extra directory)

File: nnotepad/css/index.css
Comment (inexorabletash) [inline nnotepad/css/index.css:197]: `"../common/css/fonts/victor-mono-v3-latin-500.woff2"`

should be

`"../../common/css/fonts/victor-mono-v3-latin/victor-mono-v3-latin-500.woff2"`

(note extra directory)

File: nnotepad/js/index.js
Comment (inexorabletash) [inline nnotepad/js/index.js:3]: It looks like this reference depends on using the 'vite' server (via `npm run start` in the root); is there any chance to make this an explicit reference so the site can be tested using just the lightweight `python3 -m http.server 8888` ?

File: nnotepad/js/index.js
Comment (inexorabletash) [inline nnotepad/js/index.js:94]: The addition of the header at the top of the page makes this resize calculation incorrect - if you move the resize bar it no longer tracks the mouse, but is offset by the height of the header. Can you fix? Might just be `${e.clientY - e.target.getBoundingClientRect().top}` or something.

**Review by inexorabletash — CHANGES_REQUESTED** [review]
Nice work!

Some nitpicks, some requests.

Also, I'm seeing lots of console spam, e.g.:

* Uncaught SyntaxError: Unexpected token `<`
* Error: Unexpected usage at _EditorSimpleWorker.loadForeignModule 

It's possible I've got things set up incorrectly locally, but I don't think those are coming from my code.

File: nnotepad/css/index.css
Comment (ibelem) [reply nnotepad/css/index.css]: Thanks, fixed the silly mistake.

File: nnotepad/css/index.css
Comment (ibelem) [reply nnotepad/css/index.css]: Thanks, fixed the silly mistake.

File: nnotepad/js/index.js
Comment (ibelem) [reply nnotepad/js/index.js]: Thanks, fixed this issue by 

```
document.documentElement.style.setProperty(
          '--input-height',
          `${e.clientY - $('#nav').offsetHeight}px`,
      );
```

File: nnotepad/js/index.js
Comment (ibelem) [reply nnotepad/js/index.js]: Use browser script rather tham esm import instead, to make the both "npm run start" and "python3 -m http.server 8888" work, please note that WebNN requires secure context (+https) to work.

**Comment by ibelem** [comment]
@inexorabletash @huningxin @anssiko 

Thanks for the review and feedback, removed monaco-editor and vite-plugin-monaco-editor npm packages, and fixed issues captured by @inexorabletash. PTAL.

Screenshot 2024-06-13 173749

Screenshot 2024-06-13 173800

File: nnotepad/js/index.js
Comment (inexorabletash) [reply nnotepad/js/index.js]: Thanks! It's worth noting that "localhost" is considered a secure context, even with HTTP protocol. See https://w3c.github.io/webappsec-secure-contexts/#is-origin-trustworthy

**Comment by inexorabletash** [comment]
Thanks - fixes look good. I appreciate the effort to make the simple HTTP server work! It's unfortunate to have to have a copy of the monaco editor in the source tree - is there any way to pull that in via npm to ease maintenance?

I'm on vacation this week so I'll give a quick LGTM to unblock.

**Review by inexorabletash — APPROVED** [review]
Thank you!

**Comment by ibelem** [comment]
@inexorabletash Wishing you a fantastic break! 

The new commit removed monaco editor local codes from source tree, use the js files from CDN https://cdn.jsdelivr.net/npm/monaco-editor@0.49.0/ instead. @huningxin @anssiko PTAL

Verify it worked via `python3 -m http.server 8888`
http://localhost:8888/nnotepad/index.html

Verify it worked via `npm run start`
https://10.239.115.52:5173/nnotepad/index.html

File: .eslintignore
Comment (inexorabletash) [inline .eslintignore:4]: If you add `/* global Monaco */` to the start of `nnotepad/js/index.js` then eslint passes and you can revert this change.

**Review by inexorabletash — APPROVED** [review]
One suggested change, but overall LGTM.

Thank you so much for taking this on!

File: .eslintignore
Comment (belem) [reply .eslintignore]: Thank you @inexorabletash for the great feedback.

One line code of `/* global monaco */` is much better than several `// eslint-disable-next-line no-undef`, PTAL.

Comment (ibelem) [reply .eslintignore]: Thank you @inexorabletash for the great feedback.

One line code of `/* global monaco */` is much better than several `// eslint-disable-next-line no-undef`, PTAL.

**Comment by ibelem** [comment]
@inexorabletash Thank you, added the `/* global monaco */` instead of ignoring the whole file for eslint. PTAL.

**Review by inexorabletash — APPROVED** [review]
LGTM!

**Pull request merged**
